src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java

```diff
@@ -125,6 +156,93 @@ public void insertVectorSearch() {
           .body("errors", is(nullValue()));
     }
 
+    // Test to verify vector embedding size can exceed general Array length limit
+    @Test
+    public void insertBigVectorThenSearch() {
```

Comment:
Can you add a test case for vector size error? And a test case that verifies $set and findOneAndReplace command?

Reply:
Not sure about vector size (I assume you mean passing wrong sized vector?) -- I don't think validation via `Shredder` will handle that part as it doesn't have access, and I think it needs to be done in some part.

Will look into adding ITs for `$set` and `findOneAndReplace`

Reply:
Test for $set added, looking into findOneAndReplace next.

Reply:
> Not sure about vector size (I assume you mean passing wrong sized vector?) -- I don't think validation via `Shredder` will handle that part as it doesn't have access, and I think it needs to be done in some part.

What I meant was to create a test case with $vector array size > 16K which will error?

Reply:
@maheshrajamani Ahh! Yes, that makes sense. Thank you for clarifying this!

Reply:
Added test for max vector column length -- and since it turns out this was not enforced for `createCollection()`, added code for that as well.